pkg/controllers/management/kontainerdrivermetadata/metadata_handler.go

```diff
@@ -0,0 +1,166 @@
+
+	K8sVersionWindowsSystemImages   map[string]v3.WindowsSystemImages
+	K8sVersionWindowsServiceOptions map[string]v3.KubernetesServicesOptions
+}
+
+type TickerData struct {
+	cancelFunc context.CancelFunc
+	url        string
+	interval   time.Duration
+}
+
+var tickerData *TickerData
+
+func Register(ctx context.Context, management *config.ManagementContext) {
+	mgmt := management.Management
+
+	m := &MetadataController{
+		SystemImagesLister:        mgmt.RKEK8sSystemImages("").Controller().Lister(),
+		SystemImages:              mgmt.RKEK8sSystemImages(""),
+		ServiceOptionsLister:      mgmt.RKEK8sServiceOptions("").Controller().Lister(),
+		ServiceOptions:            mgmt.RKEK8sServiceOptions(""),
+		AddonsLister:              mgmt.RKEAddons("").Controller().Lister(),
+		Addons:                    mgmt.RKEAddons(""),
+		WindowsSystemImagesLister: mgmt.RKEK8sWindowsSystemImages("").Controller().Lister(),
+		WindowsSystemImages:       mgmt.RKEK8sWindowsSystemImages(""),
+		ctx:                       ctx,
+	}
+
+	// load values on startup and start ticker
+	logrus.Infof("driverMetadata: loading data")
+	m.refresh(settings.RkeMetadataURL.Get())
+	m.initTicker(ctx)
+
+	mgmt.Settings("").AddHandler(ctx, "rke-metadata-handler", m.sync)
+}
+
+func (m *MetadataController) sync(key string, test *v3.Setting) (runtime.Object, error) {
+	if test == nil || (test.Name != "rke-metadata-url" && test.Name != "rke-metadata-refresh-interval") {
+		return nil, nil
+	}
+	url, interval, err := getSettingValues()
+	if err != nil {
+		return nil, fmt.Errorf("driverMetadata error getting interval in int %s %v", settings.RkeMetadataRefreshInterval.Get(), err)
+	}
+	if tickerData != nil {
+		if tickerData.url != url || tickerData.interval != interval {
+			tickerData.cancelFunc()
+
+			logrus.Infof("driverMetadata: starting new %s %v", url, interval)
+			cctx, cancel := context.WithCancel(m.ctx)
+			tickerData.interval = interval
+			tickerData.url = url
+			tickerData.cancelFunc = cancel
+
+			go m.startTicker(cctx, tickerData)
+		}
+	}
+	return test, nil
+}
+
+func (m *MetadataController) initTicker(ctx context.Context) {
+	url, interval, err := getSettingValues()
+	if err != nil {
+		panic(fmt.Errorf("driverMetadata error getting interval in int %s %v", settings.RkeMetadataRefreshInterval.Get(), err))
+	}
+	cctx, cancel := context.WithCancel(ctx)
+
+	tickerData = &TickerData{cancelFunc: cancel, url: url, interval: interval}
+
+	go m.startTicker(cctx, tickerData)
+}
+
+func (m *MetadataController) startTicker(ctx context.Context, tickerData *TickerData) {
+	checkInterval, url := tickerData.interval, tickerData.url
+	for range ticker.Context(ctx, checkInterval) {
+		logrus.Infof("driverMetadata: checking rke-metadata-url every %v", checkInterval)
+		m.refresh(url)
+	}
+}
+
+func (m *MetadataController) refresh(url string) {
+	data, err := loadData(url)
+	if err != nil {
+		logrus.Errorf("error loading rke data, using stored defaults %v", err)
```

Comment:
Do we need to even load anything from vendor if data has been synced once. I would say just keep using what has been synced. And not load older vendor on subsequent failures. 
- use vendor on first create only
- on every refresh pull from repo, if it fails do not reload from vendor